reviewing.md

```diff
@@ -0,0 +1,126 @@
+# Code Review in Cluster API
+
+## Goal of this document
+
+- To help newcomers to the project in implementing better PRs given the knowledge of what will be evaluated 
+  during the review.
+- To help contributors in stepping up as a reviewer given a common understanding of what are the most relevant
+  things to be evaluated during the review.
+
+IMPORTANT! Improving and maintaining this document is a collaborative effort, so we are encouraging constructive
+feedback and suggestions.
+
+   * [Code Review in Cluster API](#code-review-in-cluster-api)
+      * [Goal of this document](#goal-of-this-document)
+      * [Relevant docs and videos](#relevant-docs-and-videos)
+      * [Definition](#definition)
+      * [CAPI Design considerations](#capi-design-considerations)
+         * [Reentrancy](#reentrancy)
+         * [API design](#api-design)
+            * [Serialization](#serialization)
+            * [Owner References](#owner-references)
+         * [The Cluster API contract](#the-cluster-api-contract)
+
+## Relevant docs and videos
+
+- [Contributor Summit NA 2019: Keeping the Bar High - How to be a bad ass Code Reviewer](https://www.youtube.com/watch?v=OZVv7-o8i40)
+- [Code Review Developer Guide - Google](https://google.github.io/eng-practices/review/)
+- [The Gentle Art Of Patch Review](https://sage.thesharps.us/2014/09/01/the-gentle-art-of-patch-review/)
+
+## Definition
+
+(from [Code Review Developer Guide - Google](https://google.github.io/eng-practices/review/)) 
+
+A code review is a process where someone other than the author(s) of a piece of code examines that code
+
+Code reviews should look at:
+
+- Design: Is the code well-designed and appropriate for your system?
+- Functionality: Does the code behave as the author likely intended? Is the way the code behaves good for its users?
+- Complexity: Could the code be made simpler? Would another developer be able to easily understand and use this code
+  when they come across it in the future?
+- Tests: Does the code have correct and well-designed automated tests?
+- Naming: Did the developer choose clear names for variables, classes, methods, etc.?
+- Comments: Are the comments clear and useful?
+- Style: Does the code follow our style guides?
+- Documentation: Did the developer also update relevant documentation?
```

Comment:
Should we keep this in two places, or just link to one? It might be useful to keep it in one place so we can avoid duplication

Reply:
IMO a good split is having `CONTRIBUTING.md ` focusing on _how_ the review process works (which labels to apply, when, who), while `reviewing.md` focuses on _what_ a review should look at, so this list belongs to this second page.

But I'm open to suggestions...

Reply:
removed